src/user.js

```diff
@@ -0,0 +1,169 @@
+      this.awsCognito.CognitoUserPool, this.awsCognito.CognitoUser);
+
+    return new Promise((resolve, reject) => {
+      cognitoUser.confirmRegistration(confirmationCode, true, (err, result) => {
+        if (err) {
+          return reject(err);
+        }
+
+        return resolve(result);
+      });
+    });
+  }
+
+  resendConfirmationCode(username, userPoolId, clientId) {
+    const cognitoUser = getCognitoUser(username, userPoolId, clientId,
+      this.awsCognito.CognitoUserPool, this.awsCognito.CognitoUser);
+
+    return new Promise((resolve, reject) => {
+      cognitoUser.resendConfirmationCode((err, result) => {
+        if (err) {
+          return reject(err);
+        }
+
+        return resolve(result);
+      });
+    });
+  }
+
+  getAttributes(username, userPoolId, clientId) {
+    const cognitoUser = getCognitoUser(username, userPoolId, clientId,
+      this.awsCognito.CognitoUserPool, this.awsCognito.CognitoUser);
+
+    return new Promise((resolve, reject) => {
+      cognitoUser.getAttributes((err, result) => {
+        if (err) {
+          return reject(err);
+        }
+
+        return resolve(result);
+      });
+    });
+  }
+
+  deleteAttributes(username, attributeNames, userPoolId, clientId) {
+    const cognitoUser = getCognitoUser(username, userPoolId, clientId,
+      this.awsCognito.CognitoUserPool, this.awsCognito.CognitoUser);
+
+    return new Promise((resolve, reject) => {
+      cognitoUser.deleteAttributes(attributeNames, (err, result) => {
+        if (err) {
+          return reject(err);
+        }
+
+        return resolve(result);
+      });
+    });
+  }
+
+  updateAttributes(username, attributes, userPoolId, clientId) {
+    const cognitoUser = getCognitoUser(username, userPoolId, clientId,
+      this.awsCognito.CognitoUserPool, this.awsCognito.CognitoUser);
+    const cognitoUserAttributes = getCognitoUserAttributes(attributes,
+      this.awsCognito.CognitoUserAttribute);
+
+    return new Promise((resolve, reject) => {
+      cognitoUser.updateAttributes(cognitoUserAttributes, (err, result) => {
+        if (err) {
+          return reject(err);
+        }
+
+        return resolve(result);
+      });
+    });
+  }
+
+  changePassword(username, oldPassword, newPassword, userPoolId, clientId) {
+    const cognitoUser = getCognitoUser(username, userPoolId, clientId,
+      this.awsCognito.CognitoUserPool, this.awsCognito.CognitoUser);
+
+    return new Promise((resolve, reject) => {
+      cognitoUser.changePassword(oldPassword, newPassword, (err, result) => {
+        if (err) {
+          return reject(err);
+        }
+
+        return resolve(result);
+      });
+    });
+  }
+
+  forgotPassword(username, userPoolId, clientId) {
+    const cognitoUser = getCognitoUser(username, userPoolId, clientId,
+      this.awsCognito.CognitoUserPool, this.awsCognito.CognitoUser);
+
+    return new Promise((resolve, reject) => {
+      cognitoUser.forgotPassword({
+        onSuccess(data) {
```

Comment:
I've created a generic promise callback function for this now - removes the duplication quite a bit.